### PR TITLE
gh-93654: Add module docstring to pathlib

### LINKED ISSUE #93654: pathlib has no docstring
**Documentation**

Oddly, `pathlib` has no docstring: https://github.com/python/cpython/blob/main/Lib/pathlib.py

```
>>> import pathlib
>>> type(pathlib.__doc__)
<class 'NoneType'>
```

### PR DESCRIPTION
**Pull request by kj7rrv**
Issue: gh-93654

Automerge-Triggered-By: GH:brettcannon

### CODE REVIEW

**Comment by kj7rrv** [comment]
What is the correct way to mark up the name of a class in ReST?

**Comment by brettcannon** [comment]
/cc @barneygale 



> What is the correct way to mark up the name of a class in ReST?

I assume you're referring to your news entry, and in that instance you don't mark up the class name.

**Comment by kj7rrv** [comment]
@brettcannon 2acf6ded420c4b391629960fa13dc88d1594806b should fix it.

File: Lib/pathlib.py [hunk 1]
```diff
@@ -1,3 +1,6 @@
+"""Classes representing filesystem paths with semantics appropriate for
+different operating systems."""
```
Comment (merwok) [inline Lib/pathlib.py]:
```suggestion
"""Object-oriented filesystem paths.

This module provides classes to represent abstract paths and concrete
paths with operations that have semantics appropriate for different
operating systems.
"""
```

First line is taken from the pathlib PEP, the rest from your patch.

Please read PEP 257 for docstring conventions — docstrings should be one line, or start with one line that stands on its own.

**Comment by merwok** [comment]
✅  but I would like someone else to approve the text 🙂

**Review by barneygale — APPROVED** [review]
Looks good to me, nice work.

**Review by brettcannon — APPROVED** [review]

**Pull request merged**